languages/python/django-oso/tests/test_post_relationship.py

```diff
@@ -765,20 +790,22 @@ def test_redundant_in_on_same_field(tag_nested_many_many_fixtures):
     user = User.objects.get(username="user")
     authorize_filter = authorize_model(None, Post, actor=user, action="read")
     posts = Post.objects.filter(authorize_filter)
-    expected = """
+    expected = f"""
 
         SELECT "test_app2_post"."id", "test_app2_post"."contents", "test_app2_post"."access_level",
                "test_app2_post"."created_by_id", "test_app2_post"."needs_moderation"
         FROM "test_app2_post"
-        LEFT OUTER JOIN "test_app2_post_tags" ON ("test_app2_post"."id" = "test_app2_post_tags"."post_id")
-        WHERE (EXISTS(SELECT U0."id"
-                      FROM "test_app2_tag" U0
-                      WHERE (U0."id" = "test_app2_post_tags"."tag_id"
-                             AND U0."is_public"))
-               AND EXISTS(SELECT U0."id"
-                          FROM "test_app2_tag" U0
-                          WHERE (U0."id" = "test_app2_post_tags"."tag_id"
-                                 AND U0."name" = random)))
+        WHERE "test_app2_post"."id" IN (SELECT V0."id"
+                                        FROM "test_app2_post" V0
+                                        LEFT OUTER JOIN "test_app2_post_tags" V1 ON (V0."id" = V1."post_id")
+                                        WHERE (EXISTS(SELECT U0."id"
+                                                      FROM "test_app2_tag" U0
+                                                      WHERE (U0."id" = {parenthesize('V1."tag_id"')}
+                                                      AND U0."name" = random)){is_true()}
+                                                      AND EXISTS(SELECT U0."id"
```

Comment:
This should line up with the `EXISTS` on line 801, right?

Reply:
Maybe?

Reply:
Oh no I'm off — where it is is 👍